### ISSUE TITLE
Merge incompatible types

### ISSUE THREAD

**Issue by danthegoodman1** [mokkabonna/json-schema-merge-allof#39]
If there are schema with with types `string` and `number` being merged, is there an option to have the merged schema come out with `type: ['string', 'number']`?

**Comment by mokkabonna** [comment]
No and probably will not be. As that would change the schemas fundamentally and change the original meaning of the schema.

However you can do this yourself if you wnat that custom behaviour.

````js
  const result = merge(
    {
      type: "string",
      allOf: [
        {
          type: "number",
        },
      ],
    },
    {
      resolvers: {
        type(values, path, mergeSchemas, options) {
          return values;
        },
      },
    }
  );

  console.log(JSON.stringify(result, null, 2));
```

```json
{
  "type": [
    "string",
    "number"
  ]
}
```

You probably also want to remove any duplicates on the values.

**Comment by danthegoodman1** [comment]
Yep, that’s what I just finished doing. Thanks! However that is still valid
json schema, so I think it’s something to make more visible for folks who
want a property to be one of two or more types

**Comment by mokkabonna** [comment]
Yes multiple types are allowed and supported.

It is just that this schema will always fail validation. One schema says it must be string, another that it must be number.
```json
{
      type: "string",
      allOf: [
        {
          type: "number",
        },
      ],
}
```
This library does not combine schemas and allow all variants. It preserves the validation of how it was before.

Either both of the schemas should have `"type": ["string", "number"]` or just one of them. That way it would work as expected.

**Comment by danthegoodman1** [comment]
Copy, thanks for clarifying
